homeassistant/components/http/static.py

```diff
@@ -40,32 +45,30 @@ class CachingStaticResource(StaticResource):
     """Static Resource handler that will add cache headers."""
 
     async def _handle(self, request: Request) -> StreamResponse:
+        """Return requested file from disk as a FileResponse."""
         rel_url = request.match_info["filename"]
-        hass: HomeAssistant = request.app[KEY_HASS]
-        filename = Path(rel_url)
-        if filename.anchor:
-            # rel_url is an absolute name like
-            # /static/\\machine_name\c$ or /static/D:\path
-            # where the static dir is totally different
-            raise HTTPForbidden()
-        try:
-            key = (filename, self._directory, self._follow_symlinks)
-            if (filepath := PATH_CACHE.get(key)) is None:
-                filepath = PATH_CACHE[key] = await hass.async_add_executor_job(
-                    _get_file_path, filename, self._directory, self._follow_symlinks
-                )
-        except (ValueError, FileNotFoundError) as error:
-            # relatively safe
-            raise HTTPNotFound() from error
-        except Exception as error:
-            # perm error or other kind!
-            request.app.logger.exception(error)
-            raise HTTPNotFound() from error
+        key = (rel_url, self._directory, self._follow_symlinks)
+        if (filepath := PATH_CACHE.get(key)) is None:
+            hass: HomeAssistant = request.app[KEY_HASS]
+            try:
+                filepath = await hass.async_add_executor_job(_get_file_path, *key)
+            except (ValueError, FileNotFoundError) as error:
+                # relatively safe
+                raise HTTPNotFound() from error
+            except HTTPForbidden:
```

Comment:
Is the codecov result stale?

Reply:
Its not stale, aiohttp actually blocks it before we get to this point. Since its a security control, I adjusted the test so its tested directly in case aiohttp every does not block it anymore.

Reply:
Thanks!